### PR TITLE
check if report.icons and report.icons[0] exists

### PR DESCRIPTION
**Pull request by marcochavezf**
### Details

Context: https://github.com/Expensify/App/issues/5232

Sometimes the avatars disappear from the LHN and I was only able to reproduce the issue after logging out and loging in again with a different account. 

I found out the value of `report.actions` sometimes is `undefined` or has an array with an empty string (`['']`). This is because the links of the avatar thumbnails are fetched after inserting the `simplifiedReports` into Onyx: 

Screen Shot 2021-09-20 at 16 03 26

here ^ `PersonalDetails.getFromReportParticipants(Object.values(simplifiedReports))` is called to fetch the avatar thumbnails asynchronously (using `API.PersonalDetails_GetForEmails`), which resolves after calling
`Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT, simplifiedReports)` which triggers a re-render synchronously and calls the function `createOption` before fetching the avatars:

Screen Shot 2021-09-20 at 16 17 18

My hypothesis is that the avatar won't appear if `API.PersonalDetails_GetForEmails` doesn't resolve because of a connection issue (i.e. internet goes offline exactly when the app is fetching the avatars). So I added more conditions to check not only if the `report` object exists but also the `icons` and `icons[0]` has a valid value; I found sometimes `icons[0]` is an empty string too: https://github.com/Expensify/App/blob/main/src/libs/OptionsListUtils.js#L760 (which is called in `PersonalDetails.getFromReportParticipants`).

Seems the issue was only reproducible on Desktop and Web.

### Fixed Issues

$ https://github.com/Expensify/App/issues/5232

### Tests

1. Log in
2. Log out
3. Log in with a different account

### QA Steps

1. Log in
2. Log out
3. Log in with a different account

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web

Screen Shot 2021-09-20 at 18 35 51

#### Desktop

Screen Shot 2021-09-20 at 18 38 32

### CODE REVIEW

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -230,10 +230,11 @@ function createOption(personalDetailList, report, draftComments, {
             ? lastMessageText
             : Str.removeSMSDomain(personalDetail.login);
     }
+    const icons = (report && report.icons && report.icons[0]) ? report.icons : [personalDetail.avatar];
```
Comment (tgolen) [inline src/libs/OptionsListUtils.js:233]: Can you please use `Lodash.get` instead for this? I think it would make this more simple.

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/libs/OptionsListUtils.js (hunk 1 above)
Comment (marcochavezf) [reply src/libs/OptionsListUtils.js]: Sure!

**Comment by marcochavezf** [comment]
Update: use `lodashGet` to retrieve either report.actions or [personalDetail.avatar]

**Review by tgolen — APPROVED** [review]
Ah, that looks great. Thanks!

**Pull request merged**